src/main/java/com/gmail/llmdlio/townyflight/TownyFlightAPI.java

```diff
@@ -262,4 +308,50 @@ public static void cachePlayerFlight(Player player, boolean canFly) {
 	public static void removeCachedPlayer(Player player) {
 		canFlyCache.remove(player.getUniqueId());
 	}
+
+	// Increments the enemiesInTown counter hashmap for the given town.
+	public static void incrementEnemiesInTown(Town town) {
+
+		UUID uuid = town.getUUID();
+
+		if(enemiesInTown.containsKey(uuid)){
+			enemiesInTown.put(uuid, enemiesInTown.get(uuid) + 1);
+        }
+		else{
+			enemiesInTown.put(uuid, 1);
+        }
+        takeFlightFromPlayersInTown(town, "enemynearby");
+
+    }
+
+	// Decrements the enemiesInTown counter hashmap for the given town.
+	public static void decrementEnemiesInTown(Town town) {
+
+		UUID uuid = town.getUUID();
+
+		if(enemiesInTown.containsKey(uuid)){
+			enemiesInTown.put(uuid, enemiesInTown.get(uuid) - 1);
+
+			// Re-add flight if there are no more enemies in town.
+			if(enemiesInTown.get(uuid) <= 0){
+				addFlightToPlayersInTown(town);
+			}
+		}
+		else{
+			plugin.getServer().getLogger().severe("Tried to decrement enemies in town for a town that shouldn't have any enemies.");
+		}
+	}
+
+	// Returns true if the hashmap contains the supplied town and the value is >0.
+	public static boolean containsTown(Town town) {

```

Comment:
This method should accept a UUID instead of the whole Town.
```suggestion
	public static boolean townHasEnemiesPresent(UUID uuid) {

```